src/memote/support/consistency.py

```diff
@@ -490,37 +492,122 @@ def find_disconnected(model):
     return [met for met in model.metabolites if len(met.reactions) == 0]
 
 
-def find_metabolites_not_produced_with_open_bounds(model):
+def _init_worker(model, irr, val):
+    """
+    Initialize a global model object for multiprocessing.
+
+    Parameters
+    ----------
+    model : cobra.Model
+        The metabolic model under investigation.
+    irr: optlang.Variable || cobra.Reaction
+        the reaction to be added to the linear coefficients. It must be in the
+        variables of the model.
+    val: int
+        value of the coefficient: -1 for production and 1 for consumption
+
+    """
+    global _model
+    global _irr
+    global _val
+    _model = model
+    _model.objective = irr
+    _irr = irr
+    _val = val
+
+
+def _solve_metabolite_production(metabolite):
+    """
+    Add reaction to a `metabolite`'s contraints.
+
+    The reaction and the model are passed as globals.
+
+    Parameters
+    ----------
+    metabolite: cobra.Metabolite
+        the reaction will be added to this metabolite as a linear coefficient
+
+    Returns
+    -------
+    solution: float
+        the value of the solution of the LP problem, *NaN* if infeasible.
+    metabolite: cobra.Metabolite
+        metabolite passed as argument (to use map as a filter)
+
+    """
+    global _model
```

Comment:
As far as I can tell `_init_worker` is always invoked before `_solve_metabolite_production` hence the declaration of these global variables is redundant (they are already global). As far as I recall you don't need to declare them as global to read their values.

Reply:
Yes, you are right. Also, the changes doesn't affect speed of the multiprocessing feature, as noted in [this updated notebook](https://github.com/carrascomj/notebooks/blob/master/memote/test_consistency_prod_consumption_speed_up.ipynb).